shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.mm

```diff
@@ -102,6 +145,14 @@ - (void)performCommit:(CGSize)size notify:(nonnull dispatch_block_t)notify {
   event.Wait();
 }
 
+- (void)registerView:(int64_t)viewId {
+  _contentSizes[viewId] = CGSizeZero;
+}
+
+- (void)deregisterView:(int64_t)viewId {
+  _contentSizes.erase(viewId);
```

Comment:
Is this meant to be called from platform thread? If not then maybe it should take the lock and notify `_condBlockResize` if it is actively waiting for content.

Reply:
I've asserted that this method is called from the platform thread.